vulkan_helpers/helper_functions.cpp

```diff
@@ -1154,8 +1154,9 @@ VkSwapchainKHR CreateDefaultSwapchain(
         extensions,                                   // pNext
         flags,                                        // flags
         *surface,                                     // surface
-        std::min(surface_caps.minImageCount + 1,
-                 maxSwapchains),        // minImageCount
+        (min_image_count == 0) ? std::min(
```

Comment:
This doesn't do what you want.
This sets the number of images to EXACTLY min_image_count, if it is set.
The logic here has to be a bit more explicit.

We need to have at LEAST min_image_count, at LEAST surface_caps.minImageCount, and at MOST surface_caps. maxImageCount.
These may also be mutually exclusive. Meaning we need an error case.